flang/include/flang/Lower/PFTBuilder.h

```diff
@@ -360,61 +365,155 @@ struct ProgramUnit : ProgramVariant {
 /// Fortran EQUIVALENCE statements are a mechanism that introduces aliasing
 /// between named variables. The set of overlapping aliases will materialize a
 /// generic store object with a designated offset and size. Participant
-/// symbols will simply be pointers into the primary store.
+/// symbols will simply be pointers into the aggregate store.
+///
+/// EQUIVALENCE can also interact with COMMON and other global variables to
+/// imply aliasing between (subparts of) a global and other local variable
+/// names.
 ///
 /// Properties can be applied by lowering. For example, a local array that is
 /// known to be very large may be transformed into a heap allocated entity by
 /// lowering. That decision would be tracked in its Variable instance.
 struct Variable {
-  using StoreInterval = std::tuple<std::size_t, std::size_t>;
+  /// Most variables are nominal and require the allocation of local/global
+  /// storage space. A nominal variable may also be an alias for some other
+  /// (subpart) of storage.
+  struct Nominal {
+    Nominal(const semantics::Symbol *symbol, int depth, bool global)
+        : symbol{symbol}, depth{depth}, global{global} {}
+    const semantics::Symbol *symbol{};
+    int depth{};
+    bool global{};
+    bool heapAlloc{}; // variable needs deallocation on exit
+    bool pointer{};
+    bool target{};
+    bool aliaser{}; // participates in EQUIVALENCE union
+    std::size_t aliasOffset{};
+  };
+
+  using Interval = std::tuple<std::size_t, std::size_t>;
```

Comment:
Just curious - why std::tuple, rather than std::pair?

Reply:
> Use of the term "interval" for storage sets doesn't match so well. Maybe consider alternatives, possibly aggregateSet, or equivalenceSet, or storageSet, or ???

Since the algorithm is identifying overlapping intervals of offsets, calling them intervals seems apt.

Reply:
I'm not sure I'll keep the tuple around. This code has been evolving quite a bit with multiple implementations, etc. There is more refactoring needed. Particularly since the front-end may change how EQUIVALENCE with initialized globals is presented soon.